x-pack/plugins/apm/public/components/app/correlations/latency_correlations.tsx

```diff
@@ -256,17 +253,48 @@ export function LatencyCorrelations({ onFilter }: { onFilter: () => void }) {
     [history, onFilter, trackApmEvent]
   );
 
-  const histogramTerms: MlCorrelationsTerms[] = useMemo(() => {
-    return histograms.map((d) => {
-      return {
-        fieldName: d.field,
-        fieldValue: d.value,
-        ksTest: d.ksTest,
-        correlation: d.correlation,
-        duplicatedFields: d.duplicatedFields,
-      };
-    });
-  }, [histograms]);
+  const [sortField, setSortField] = useState<keyof MlCorrelationsTerms>(
+    'correlation'
+  );
+  const [sortDirection, setSortDirection] = useState<Direction>('desc');
+
+  const onTableChange = useCallback(({ sort }) => {
+    const { field: currentSortField, direction: currentSortDirection } = sort;
+
+    setSortField(currentSortField);
+    setSortDirection(currentSortDirection);
+  }, []);
+
+  const { histogramTerms, sorting } = useMemo(() => {
+    if (!Array.isArray(histograms)) {
+      return { histogramTerms: [], sorting: undefined };
+    }
+    const sortedTerms = sortBy(
+      histograms.map((d) => {
+        return {
+          fieldName: d.field,
+          fieldValue: d.value,
+          ksTest: d.ksTest,
+          correlation: d.correlation,
+          duplicatedFields: d.duplicatedFields,
+        };
+      }),
+      sortField
+    );
+
+    return {
+      histogramTerms:
+        sortDirection === 'asc'
+          ? sortedTerms
+          : (sortedTerms.reverse() as MlCorrelationsTerms[]),
```

Comment:
You can use `orderBy` instead:

```suggestion
    const sortedTerms = orderBy(
      histograms.map((d) => {
        return {
          fieldName: d.field,
          fieldValue: d.value,
          ksTest: d.ksTest,
          correlation: d.correlation,
          duplicatedFields: d.duplicatedFields,
        };
      }),
      sortField,
      sortDirection
    );

    return {
      histogramTerms: sortedTerms,
```

Reply:
Updated here https://github.com/elastic/kibana/pull/109187/commits/82d5457f4696584969059f9ad78a897722507d2c